administrator/components/com_menus/src/Field/MenuItemByComponentField.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/**
+ * @package     Joomla.Administrator
+ * @subpackage  com_menus
+ *
+ * @copyright   (C) 2022 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+namespace Joomla\Component\Menus\Administrator\Field;
+
+use Joomla\CMS\Factory;
+use Joomla\CMS\Form\Field\ListField;
+use Joomla\CMS\Language\Text;
+use Joomla\Database\ParameterType;
+use Joomla\Utilities\ArrayHelper;
+
+/**
+ * MenuItem by Component field.
+ *
+ * @since __DEPLOY_VERSION__
+ */
+class MenuItemByComponentField extends ListField
+{
+    /**
+     * The form field type.
+     *
+     * @var     string
+     * @since   __DEPLOY_VERSION__
+     */
+    protected $type = 'MenuItemByComponent';
+
+    /**
+     * Method to get a list of options for a list input.
+     *
+     * @return    array  An array of JHtml options.
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    protected function getOptions()
+    {
+        // Initialise variable.
+        $db      = Factory::getDbo();
+        $options = [];
+
+        $query = $db->getQuery(true);
+        $query->select('DISTINCT ' . $db->quoteName('extensions.element'))
+            ->from($db->quoteName('#__menu', 'menu'))
+            ->join(
+                'INNER',
+                $db->quoteName('#__extensions', 'extensions'),
+                $db->quoteName('extensions.extension_id') . ' = ' . $db->quoteName('menu.component_id')
+            )
+            ->where($db->quoteName('menu.client_id') . ' = 0')
+            ->where($db->quoteName('menu.type') . ' = ' . $db->quote('component'))
+            ->where($db->quoteName('extensions.extension_id') . ' IS NOT NULL');
+
+        $app             = Factory::getApplication();
+        $currentMenuType = $app->input->getString('menutype', $app->getUserState($this->context . '.menutype', ''));
```

Comment:
```suggestion
        $currentMenuType = $app->getInput()->getString('menutype', $app->getUserState($this->context . '.menutype', ''));
```
See https://github.com/joomla/joomla-cms/pull/39029